CalcualtorLogic.js

```diff
@@ -152,21 +152,21 @@ class Calculator {
         div6.appendChild(clearingButton);
         wrapperDiv.appendChild(div6);
         this.hostElem.appendChild(wrapperDiv);
+        const div1 = this.getButtonsRow(["7", "8", "9"]);
+        const div2 = this.getButtonsRow(["4", "5", "6"]);
+        const div3 = this.getButtonsRow(["1", "2", "3"]);
     };
     getButtonsRow(buttons) {
         const div = document.createElement("div");
         for (let button of buttons) {
-            button = document.createElement("button");
-            button.dataset.number;
-            const buttonText = document.createTextNode();
-            button.appendChild(buttonText);
-            div.appendChild(button);
+            var buttonElement = document.createElement("button");
+            buttonElement.dataset.number = this.getButtonsRow(["7", "8", "9"]);
```

Comment:
This incorrect.
`buttonElement.dataset.number`  should be giving a string value which is the button value 
you're looping over the these values `["7", "8", "9"]`  suing the `for of ` loop on line 161 
and you have `button` variable waiting to be used correctly

Reply:
would you let me know what is the task of the button in the line 161 
thanks